src/ansys/mapdl/core/hpc/pyhps.py

```diff
@@ -73,487 +73,846 @@ def get_value_from_json_or_default(
+
+        if extra_files and not all([os.path.exists(each) for each in extra_files]):
+            raise ValueError("One or more extra files does not exist.")
+
+        return extra_files
+
+    def _validate_output_files(self, output_files):
+        """Validate output_files input"""
+        if not output_files:
+            output_files = []
+        elif isinstance(output_files, str):
+            output_files = output_files.split(",")
+
+        return output_files
+
+    def _validate_requirements_file(self, requirements_file):
+        """Validate requirements_file input"""
+        return requirements_file
+
+    def _validate_python(self, python):
+        """Validate python inputs"""
+        if python not in [2, 2.7, 3, 3.5, 3.6, 3.7, 3.8, 3.9, 3.10, 3.11, 3.12]:
+            warn(f"Version of Python {python} might not be supported by the cluster.")
+        return python
+
+    def _validate_num_cores(self, num_cores: int):
+        """Validate num_cores inputs"""
+        if not num_cores:
+            num_cores = 0
+
+        return int(num_cores)
+
+    def _validate_memory(self, memory: int):
+        """Validate memory inputs"""
+        if not memory:
+            memory = 0
+
+        return int(memory)
+
+    def _validate_disk_space(self, disk_space: float):
+        """Validate disk_space inputs"""
+        if not disk_space:
+            disk_space = 0
+
+        return float(disk_space)
+
+    def _validate_exclusive(self, exclusive: bool):
+        """Validate exclusive inputs"""
+        if exclusive is None:
+            exclusive = False
+
+        return bool(exclusive)
+
+    def _validate_max_execution_time(self, max_execution_time):
+        """Validate max_execution_time inputs"""
+        if not max_execution_time:
+            max_execution_time = 0
+
+        return float(max_execution_time)
+
+    def _validate_name(self, name):
+        """Validate name inputs"""
+        if name is None:
+            if self.mode == "python":
+                name = "My PyMAPDL project"
+            elif self.mode == "apdl":
+                name = "My APDL project"
+            elif self.mode == "shell":
+                name = "My shell project"
+            else:
+                name = "My project"
+        return name
+
+    def _validate_mode(self, mode):
+        _, file_extension = os.path.splitext(self._main_file)
+
+        if mode is None:
+            if file_extension.lower() in [".sh"]:
+                mode = "shell"
+            elif file_extension.lower() in [".py"]:
+                mode = "python"
+            elif file_extension.lower() in [".inp", ".mac"]:
+                mode = "apdl"
         else:
-            parm = StringParameterDefinition(
-                name=name,
-                display_text=name,
-                default=value,
-            )
+            if mode.lower() not in ["python", "shell", "apdl"]:
+                raise Exception("File type not supported.")
```

Comment:
```suggestion
                raise Exception("File type is not supported.")
```
Message must have a verb.